apps/examples/src/examples/speech-bubble/SpeechBubble/SpeechBubbleUtil.tsx

```diff
@@ -97,60 +98,68 @@ export class SpeechBubbleUtil extends ShapeUtil<SpeechBubbleShape> {
 	}
 
 	override getHandles(shape: SpeechBubbleShape) {
-		const handles = getHandleinShapeSpace(shape).props.handles
-		const handlesArray = Object.values(handles)
-
-		return handlesArray
+		const {
+			handles: { handle },
+			w,
+			h,
+		} = shape.props
+
+		return [
+			{
+				...handle,
+				// props.handles.handle coordinates are normalized
+				// but here we need them in shape space
+				x: handle.x * w,
+				y: handle.y * h,
+			},
+		]
 	}
 
 	// [4]
 	override onBeforeUpdate: TLOnBeforeUpdateHandler<SpeechBubbleShape> | undefined = (
 		_: SpeechBubbleShape,
 		next: SpeechBubbleShape
 	) => {
-		const shape = getHandleinShapeSpace(next)
+		const { w, h, handles } = next.props
 
-		const { originalIntersection: intersection, insideShape } = getHandleIntersectionPoint({
-			w: shape.props.w,
-			h: shape.props.h,
-			handle: shape.props.handles.handle,
-		})
-
-		if (!intersection) throw new Error('No intersection')
+		const { segmentsIntersection, insideShape } = getHandleIntersectionPoint(next)
```

Comment:
Easier to pass the shape in as props, not that we don't have to worry about what coordinate space its handles are in. It's cheap to compute the handle in shape space so we'll just do it where we need it.